extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java

```diff
@@ -859,6 +859,33 @@ public enum ResponseMode {
         @ConfigItem(defaultValue = "true")
         public boolean allowMultipleCodeFlows = true;
 
+        /**
+         * Fail with the HTTP 401 error if the state cookie is present but no state query parameter is present.
+         * <p/>
+         * When either multiple authentications are disabled or the redirect URL
+         * matches the original request URL, the stale state cookie might remain in the browser cache from
+         * the earlier failed redirect to an OpenId Connect provider and be visible during the current request.
+         * For example, if Single-page application (SPA) uses XHR to handle redirects to the provider
+         * which does not support CORS for its authorization endpoint, the browser will block it
+         * and the state cookie created by Quarkus will remain in the browser cache.
+         * Quarkus will report an authentication failure when it will detect such an old state cookie but find no matching state
+         * query parameter.
+         * <p/>
+         * Reporting HTTP 401 error is usually the right thing to do in such cases, it will minimize a risk of the
+         * browser redirect loop but also can identify problems in the way SPA or Quarkus application manage redirects.
+         * For example, enabling {@link #javaScriptAutoRedirect} or having the provider redirect to URL configured
+         * with {@link #redirectPath} may be needed to avoid such errors.
+         * <p/>
+         * However, setting this property to `false` may help if the above options are not suitable.
+         * It will cause a new authentication redirect to OpenId Connect provider.
+         * <p/>
+         * Make sure it won't cause browser redirect loops if you decide to do so.
+         * Please keep in mind that unmatched state cookies will have to be removed.
```

Comment:
By this, you probably mean that Quarkus will remove all state cookies when `failOnMissingStateParam` is `false` and there is zero or many state query params, right? Before I read the code, I thought that client need to keep in mind that response will still contain state cookies. Is sentence clear to you @sberyozkin , please?

Reply:
Hmm, maybe I wasn't very clear. What I mean is that I would add removed _by whom_.

Reply:
Indeed, it should be `Please keep in mind that unmatched state cookies will have to be removed if this property is set to false`

>and there is zero or many state query params

At the moment this PR is only about supporting a case where a request arrives, we have a state cookie, but no `state` query parameter. At the moment it is an immediate `401` but I'd like to give users an option to ignore that the state cookie has not been matched and just request reauthentication - and because the state cookie is not matched, it has to be cleared.

The reason I add the text cautioning about it is, if we have a multi-tab authentication, example, an authentication is in process in one tab, user is challenged to authenticate by Keycloak, does not complete it, then opens another tab, a new request is issued, here we have another state cookie. But this multi-tab authentication is not even supported right now, without this PR, it works only if the request URL and the callback URLs are different.

So I'll probably just remove this caution to avoid the confusion, I've nearly got confused myself while typing the above text :-)

Reply:
Alright, I understand now. Thank you